src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorEngine.java

```diff
@@ -0,0 +1,296 @@
+        final Set<VCFHeaderLine> descriptions = new HashSet<>();
+
+        for ( final InfoFieldAnnotation annotation : infoAnnotations) {
+            descriptions.addAll(annotation.getDescriptions());
+        }
+        for ( final GenotypeAnnotation annotation : genotypeAnnotations) {
+            descriptions.addAll(annotation.getDescriptions());
+        }
+        for ( final String db : variantOverlapAnnotator.getOverlapNames() ) {
+            if ( VCFStandardHeaderLines.getInfoLine(db, false) != null ) {
+                descriptions.add(VCFStandardHeaderLines.getInfoLine(db));
+            } else {
+                descriptions.add(new VCFInfoHeaderLine(db, 0, VCFHeaderLineType.Flag, db + " Membership"));
+            }
+        }
+
+        return descriptions;
+    }
+
+    /**
+     * Annotates the given variant context - adds all annotations that satisfy the predicate.
+     * @param features context containing the features that overlap the given variant
+     * @param ref the reference context of the variant to annotate or null if there is none
+     * @param vc the variant context to annotate
+     * @param stratifiedPerReadAlleleLikelihoodMap par-sample map of likelihoods per read. May be null
+     * @param addAnnot function that indicates if the given annotation type should be added to the variant
+     */
+    public VariantContext annotateContext(final FeatureContext features,
+                                          final ReferenceContext ref,
+                                          final VariantContext vc,
+                                          final Map<String, PerReadAlleleLikelihoodMap> stratifiedPerReadAlleleLikelihoodMap,
+                                          final Predicate<VariantAnnotation> addAnnot) {
+        Utils.nonNull(features, "features cannot be null");
+        Utils.nonNull(vc, "vc cannot be null");
+
+        // annotate genotypes, creating another new VC in the process
+        final VariantContextBuilder builder = new VariantContextBuilder(vc);
+        builder.genotypes(annotateGenotypes(ref, vc, stratifiedPerReadAlleleLikelihoodMap, addAnnot));
+        final VariantContext newGenotypeAnnotatedVC = builder.make();
+
+        final Map<String, Object> infoAnnotMap = new LinkedHashMap<>(newGenotypeAnnotatedVC.getAttributes());
+        for ( final InfoFieldAnnotation annotationType : this.infoAnnotations) {
+            if (addAnnot.test(annotationType)){
+                final Map<String, Object> annotationsFromCurrentType = annotationType.annotate(ref, newGenotypeAnnotatedVC, stratifiedPerReadAlleleLikelihoodMap);
+                if ( annotationsFromCurrentType != null ) {
+                    infoAnnotMap.putAll(annotationsFromCurrentType);
+                }
+            }
+        }
+
+        // create a new VC in the with info and genotype annotations
+        final VariantContext annotated = builder.attributes(infoAnnotMap).make();
```

Comment:
i dont think this is a big deal - it's the same as in gatk3. if it comes up on a profiler, we'll fix it